src/app/shared/entry/GA4GHFiles.state.service.ts

```diff
@@ -0,0 +1,123 @@
+  constructor(private ga4ghService: GA4GHService) {
+    this.toolFiles$ = combineLatest(this.cwlToolFiles$, this.wdlToolFiles$, this.nflToolFiles$,
+      (cwlToolFiles, wdlToolFiles, nflToolFiles) => {
+        return this.mergeArray(cwlToolFiles, wdlToolFiles, nflToolFiles);
+      });
+    this.containerToolFiles$ = this.toolFiles$.pipe(map((toolFiles: Array<ToolFile>) => {
+      if (!toolFiles) {
+        toolFiles = [];
+      }
+      return toolFiles.filter((toolFile: ToolFile) => {
+        return toolFile.file_type === ToolFile.FileTypeEnum.CONTAINERFILE;
+      });
+    }));
+    this.descriptorToolFiles$ = this.toolFiles$.pipe(map((toolFiles: Array<ToolFile>) => {
+      if (!toolFiles) {
+        toolFiles = [];
+      }
+      return toolFiles.filter((toolFile: ToolFile) => {
+        return (toolFile.file_type === ToolFile.FileTypeEnum.PRIMARYDESCRIPTOR ||
+          toolFile.file_type === ToolFile.FileTypeEnum.SECONDARYDESCRIPTOR);
+      });
+    }));
+    this.testToolFiles$ = this.toolFiles$.pipe(map((toolFiles: Array<ToolFile>) => {
+      if (!toolFiles) {
+        toolFiles = [];
+      }
+      return toolFiles.filter((toolFile: ToolFile) => {
+        return toolFile.file_type === ToolFile.FileTypeEnum.TESTFILE;
+      });
+    }));
+  }
+
+  /**
+   * Workaround.
+   * Since the swagger.yaml does not indicate the endpoints are optionally authenticated,
+   * the generated classes will not try and use authentication.  This manually injects it in for use.
+   *
+   * @param {GA4GHService} ga4ghService
+   * @memberof GA4GHFilesStateService
+   */
+  public injectAuthorizationToken(ga4ghService: GA4GHService) {
+    ga4ghService.defaultHeaders = ga4ghService.defaultHeaders.set('Authorization',
+    ga4ghService.configuration.apiKeys['Authorization']);
+  }
+
+  update(id: string, version: string) {
+    this.injectAuthorizationToken(this.ga4ghService);
+    this.ga4ghService.toolsIdVersionsVersionIdTypeFilesGet(
+      DescriptorType.CWL, id, version).subscribe(files => {
+        this.cwlToolFiles$.next(files);
+      });
+    this.ga4ghService.toolsIdVersionsVersionIdTypeFilesGet(
+      DescriptorType.WDL, id, version).subscribe(files => {
+        this.wdlToolFiles$.next(files);
+      });
+    this.ga4ghService.toolsIdVersionsVersionIdTypeFilesGet(
+      DescriptorType.NFL, id, version).subscribe(files => {
+        this.nflToolFiles$.next(files);
+      });
+    // TODO: Grab from all descriptor types (CWL, WDL, NFL) to get all test parameter files
+
+  }
+
+  /**
+   * This merges the 3 ToolFile arrays together into one array with no duplicates
+   *
+   * @private
+   * @param {Array<ToolFile>} array1
+   * @param {Array<ToolFile>} array2
+   * @param {Array<toolFile>} array3
+   * @returns {Array<ToolFile>}
+   * @memberof GA4GHFilesStateService
+   */
+  private mergeArray(array1: Array<ToolFile>, array2: Array<ToolFile>, array3: Array<ToolFile>): Array<ToolFile> {
```

Comment:
Can probably do a varargs and make this useful for any number of arrays in the future
https://stackoverflow.com/questions/12697275/open-ended-function-arguments-with-typescript
Alternatively, library function?